src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -372,52 +372,55 @@ s32 func_80BA9110(EnBaba* this, GlobalContext* globalCtx, ScheduleResult* arg2)
 }
 
 s32 func_80BA9160(EnBaba* this, GlobalContext* globalCtx) {
-    f32 sp7C[265];
+    f32 weightArray[265];
     Vec3f sp70;
     Vec3f sp64;
-    Vec3f sp58;
+    Vec3f timePathTargetPos;
     s32 sp54 = 0;
     s32 sp50 = 0;
     s32 pad;
 
-    func_8013AF00(sp7C, 3, this->unk_410->count + 3);
+    SubS_TimePathing_FillWeightArray(weightArray, SUBS_TIME_PATHING_ORDER,
+                                     this->timePath->count + SUBS_TIME_PATHING_ORDER);
 
     if (this->unk_438 == 0) {
-        sp58 = gZeroVec3f;
+        timePathTargetPos = gZeroVec3f;
 
-        func_8013B6B0(this->unk_410, &this->unk_420, &this->unk_430, this->unk_428, this->unk_424, &this->unk_42C, sp7C,
-                      &sp58, this->unk_436);
-        func_8013B878(globalCtx, this->unk_410, this->unk_42C, &sp58);
-        this->actor.world.pos.y = sp58.y;
+        SubS_TimePathing_Update(this->timePath, &this->timePathProgress, &this->timePathElapsedTime,
+                                this->timePathWaypointTime, this->timePathTotalTime, &this->timePathWaypoint,
+                                weightArray, &timePathTargetPos, this->timePathTimeSpeed);
+        SubS_TimePathing_ComputeInitialY(globalCtx, this->timePath, this->timePathWaypoint, &timePathTargetPos);
```

Comment:
Is the `Initial` part of `SubS_TimePathing_ComputeInitialY` appropriate? It seems to be running every frame along with update, instead of only once when Initialized. What about `SubS_TimePathing_ComputeTargetPosY`? (To parallel `SubS_TimePathing_ComputeTargetPosXZ`)

Reply:
It actually only gets called once. Notice unk_438 gets set to 1 at the end of this branch, so this `if` doesn't get taken again. Its clearer in my Baba PR where I name `unk_438` `isTimePathSetup`.

Reply:
Oh you're right, I think I noticed that too but forgot on second glance